onnx/defs/shape_inference.h

```diff
@@ -13,6 +13,21 @@ namespace ONNX_NAMESPACE {
 
 using Dim = TensorShapeProto_Dimension;
 
+struct ShapeInferenceOptions {
+  // Checks the type-equality for input and output
+  bool check_type;
+  // 1: Stricter shape inference; will throw errors if any
+  // 0: Simply stop if any error
+  int strict_error_mode;
```

Comment:
I suggest change this to error_mode and let the value determine whether it is strict or something else
per current behavior : when the mode is strict aka 1 shape inf will throw any node level shape inf errors whereas when it is 0 it does not.... other errors like merging existing shape with inferred etc are thrown regardless...